mlflow/store/model_registry/sqlalchemy_store.py

```diff
@@ -247,9 +272,14 @@ def search_registered_models(self, filter_string):
         with self.ManagedSessionMaker() as session:
             if self.db_type == SQLITE:
                 session.execute("PRAGMA case_sensitive_like = true;")
-            sql_registered_models = session.query(SqlRegisteredModel).filter(*conditions).all()
+            sql_registered_models = session.query(SqlRegisteredModel)\
+                .filter(*conditions)\
+                .order_by(SqlRegisteredModel.name.asc())\
+                .offset(offset)\
```

Comment:
is `offset(0)` is the same as not applying the offset. Or is it better to conditionally apply that clause?

```
query = session.query(SqlRegisteredModel).filter(*conditions).order_by(SqlRegisteredModel.name.asc()).limit(max_results)
if page_token:
  query = query.offset(SearchUtils.parse_start_offset_from_page_token(page_token))
sql_registered_models = query.all()
```

Is there a functional / performance difference one way or another?

Reply:
There doesn't appear to be a functional difference, but I can't find anything definitive about a performance impact. Regardless, it seems reasonable to not apply the offset if the offset isn't attainable, so I changed it to do that.

Notably, I didn't take your entire suggestion, since we use the offset variable to compute the `next_page_token`, so I'd rather still call the parsing function every time.